core-bundle/assets/controllers/colorscheme-controller.js

```diff
@@ -0,0 +1,61 @@
+import { Controller } from '@hotwired/stimulus';
+
+const prefersDark = () => {
+    const prefersDark = localStorage.getItem('contao--prefers-dark');
+
+    if (null === prefersDark) {
+        return !!window.matchMedia('(prefers-color-scheme: dark)').matches;
+    }
+
+    return prefersDark === 'true';
+}
+
+export default class extends Controller {
+
+    static targets = ['label'];
+    static values = {
+        i18n: {
+            type: Object,
+            default: { light: 'Disable dark mode', dark: 'Enable dark mode'}
+        }
+    };
+
+    static afterLoad (_identifier, _application) {
+        document.firstElementChild.dataset.colorScheme = prefersDark() ? 'dark' : 'light';
```

Comment:
Any reason you are not using the usual `document.documentElement` here?

```suggestion
        document.documentElement.dataset.colorScheme = prefersDark() ? 'dark' : 'light';
```

Reply:
nope 😂  changed in e5ceb2fb7689609127c7e7f38d549432b2bca5ff